router/core/timout_transport.go

```diff
@@ -0,0 +1,44 @@
+package core
```

Comment:
Can we put this into a seperate package?

Reply:
We can't, because it needs access to the requestContext which isn't exported. If I export it, then I can put it in a different package

Reply:
I was expecting that. Thanks.